Async hooks rename

AndreasMadsen opened the pull request:
##### Checklist
- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)
async_hooks

Rename `init` to `emitInitNative` and all `emit*S` to `emit*Script`. I know some are against these purely stylish changes so I included a small consistency fix to `AsyncResource` and `emitInit` that is much more obvious with the renames.

AndreasMadsen commented on test/async-hooks/test-embedder.api.async-resource.js:
```diff
@@ -18,8 +18,8 @@ assert.throws(() => new AsyncResource('invalid_trigger_id', null),
               /^RangeError: triggerAsyncId must be an unsigned integer$/);
 
 assert.strictEqual(
-    typeof new AsyncResource('default_trigger_id').triggerAsyncId(),
```
This change wasn't needed. But when I looked for the `AsyncResource('default_trigger_id')` test, I found that it could be made more strict.

refack commented on lib/async_hooks.js:
```diff
@@ -207,42 +207,30 @@ function triggerAsyncId() {
-
-    if (typeof type !== 'string' || type.length <= 0)
-      throw new TypeError('type must be a string with length > 0');
+  constructor(type, triggerAsyncId = initTriggerId()) {
```
So just an FYI since this effectively changes the signature of the ctor (now `AsyncResource.length == 1` it would have been `semver-major`, but since this congruent with [the docs](https://nodejs.org/api/async_hooks.html#async_hooks_asyncresource_type_triggerid), I'd call this a bug fix.

trevnorris replied:
Also this is still officially experimental, which should give us more flexibility.

refack commented on test/async-hooks/test-embedder.api.async-resource.js:
```diff
@@ -18,8 +18,8 @@ assert.throws(() => new AsyncResource('invalid_trigger_id', null),
+    new AsyncResource('default_trigger_id').triggerAsyncId(),
+    async_hooks.executionAsyncId()
 );
 
```
Could you add
```js
assert.strinctEqual(AsyncResource.length, 1);
```

addaleax commented:
I’m strongly against considering changes semver-major because they modify the value of `fn.length`, experimental API or not, unless we have an actual reason to expect breakage.

refack commented:
> I’m strongly against considering changes semver-major because they modify the value of fn.length, experimental API or not, unless we have an actual reason to expect breakage.

I agree, but some do consider it as such 🤷‍♂️ Should bring it up with the release team or the CTC.

AndreasMadsen commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/9082/

AndreasMadsen commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/9087/

AndreasMadsen commented:
There were a huge amount of unrelated CI failures. It looks like most of it have been fixed in master now. I have rebased on master, hopefully that will fix it.

CI: https://ci.nodejs.org/job/node-test-pull-request/9091/

trevnorris commented on test/parallel/test-async-hooks-asyncresource-constructor.js:
```diff
@@ -4,8 +4,14 @@ require('../common');
 
 const assert = require('assert');
+const async_hooks = require('async_hooks');
 const AsyncResource = require('async_hooks').AsyncResource;
```
nit: this looks a little odd syntactically. maybe something like:
```js
const async_hooks = require('async_hooks');
const { AsyncResource } = async_hooks;
```

AndreasMadsen commented:
> Please don't forget to squash the two commits, and maybe add a message body with more explanation.

Really, you want the two commits squashed?

trevnorris commented:
@AndreasMadsen No. It's a kind reminder to make sure the commit
> [squash] enable hooks when testing asserts

is squashed into whatever commit it belongs. It's also not necessarily a reminder for you, but whomever it is that lands the PR.

AndreasMadsen commented:
I fixed the nit and added some more text in the commit message. If the CI is green I will just go ahead and merge.

CI: https://ci.nodejs.org/job/node-test-pull-request/9105/

AndreasMadsen commented:
Fixed a small merge conflict.

CI: https://ci.nodejs.org/job/node-test-pull-request/9107/

AndreasMadsen commented:
landed in 628485ea01bb1773fc8814c0d61d06fef0e12ace and 31417b68822abeced64eeb120f064d37bc133f7c

trevnorris commented:
@AndreasMadsen Thanks for the two thorough git commit messages.
